xonsh/environ.py

```diff
@@ -201,13 +206,32 @@ def wrapper(*args, **kwargs):
         return func(*args, **kwargs)
     return wrapper
 
+def ensure_hg(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        kwargs['cwd'] = kwargs.get('cwd', _get_cwd())
+        if kwargs['cwd'] is None:
+            return
+
+        # bail if hglib is not installed
+        if hglib is None:
+            return
+
+        try:
+            c = hglib.open(kwargs['cwd'])
+            c.tip()
+        except OSError:
+            # hglib is installed but hg isn't, so bail
+            return
+        except hglib.error.CommandError:
```

Comment:
Full traceback in case it is helpful:
```
Traceback (most recent call last):
  File "/usr/local/bin/xonsh", line 3, in <module>
    main()
  File "/usr/local/lib/python3.4/site-packages/xonsh/main.py", line 78, in main
    shell.cmdloop()
  File "/usr/local/lib/python3.4/site-packages/xonsh/readline_shell.py", line 114, in cmdloop
    super().cmdloop(intro=intro)
  File "/usr/local/lib/python3.4/cmd.py", line 126, in cmdloop
    line = input(self.prompt)
  File "/usr/local/lib/python3.4/site-packages/xonsh/readline_shell.py", line 127, in prompt
    return super().prompt
  File "/usr/local/lib/python3.4/site-packages/xonsh/base_shell.py", line 105, in prompt
    p = format_prompt(p)
  File "/usr/local/lib/python3.4/site-packages/xonsh/environ.py", line 366, in format_prompt
    for (k, v) in fmt.items()
  File "/usr/local/lib/python3.4/site-packages/xonsh/environ.py", line 367, in <dictcomp>
    if k in included_names}
  File "/usr/local/lib/python3.4/site-packages/xonsh/environ.py", line 290, in current_branch
    branch = get_git_branch() or get_hg_branch()
  File "/usr/local/lib/python3.4/site-packages/xonsh/environ.py", line 221, in wrapper
    c = hglib.open(kwargs['cwd'])
  File "/usr/local/lib/python3.4/site-packages/hglib/__init__.py", line 11, in open
    return client.hgclient(path, encoding, configs)
  File "/usr/local/lib/python3.4/site-packages/hglib/client.py", line 63, in __init__
    self.open()
  File "/usr/local/lib/python3.4/site-packages/hglib/client.py", line 193, in open
    self._readhello()
  File "/usr/local/lib/python3.4/site-packages/hglib/client.py", line 75, in _readhello
    ch, msg = self._readchannel()
  File "/usr/local/lib/python3.4/site-packages/hglib/client.py", line 99, in _readchannel
    raise error.ServerError()
hglib.error.ServerError
```